### PR TITLE
Enforce pre-commit to use a recent and fixed version of isort.

### PR DESCRIPTION
**Pull request by arnaudgelas**
### Before submitting

- [x]  Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x]  Did you read the contributor guideline, Pull Request section?
- [ ]  Did you make sure to update the docs?
- [ ]  Did you write any new necessary tests?
- [ ]  If you made a notable change (that affects users), did you update the CHANGELOG?

### What does this PR do?

Relates to #4805
Enforce pre-commit to use a recent and fixed version of isort rather than an uncontrolled one.

### PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### Did you have fun?

Make sure you had fun coding upside_down_face

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -22,13 +22,11 @@ repos:
       - id: trailing-whitespace
       - id: end-of-file-fixer
 
-  - repo: local
+  - repo: https://github.com/PyCQA/isort.git
```
Comment (carmocca) [inline .pre-commit-config.yaml:25]:
```suggestion
  - repo: https://github.com/PyCQA/isort
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -22,13 +22,11 @@ repos:
...
+    rev: 5.7.0
     hooks:
       - id: isort
-        name: isort
-        entry: python -m isort
         args: [--settings-path, ./pyproject.toml]
-        language: system
         types: [python]
```
Comment (carmocca) [inline .pre-commit-config.yaml:30]: this probably can be removed too, can you check?

**Comment by Borda** [comment]
@arnaudgelas pls keep in your PR the bullet list for checking...
@akihironitta @PyTorchLightning/core-contributors can you check that it works now?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (arnaudgelas) [reply .pre-commit-config.yaml]: applied

File: .pre-commit-config.yaml (hunk 2 above)
Comment (arnaudgelas) [reply .pre-commit-config.yaml]: applied

**Comment by arnaudgelas** [comment]
@Borda I have added back the checklist

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**